SC_redesign/Crop_and_Soil/soil/layer_data.py

```diff
@@ -0,0 +1,27 @@
+from dataclasses import dataclass
+from typing import Optional
+
+"""
+Each instance of this class represents a layer of soil. Each SoilData object should contain a list of LayerData objects 
+to represent its soil
+"""
+
+
+@dataclass
+class LayerData:
+    top_depth: Optional[float] = None
+    """top depth of the layer (mm)"""
+    bottom_depth: Optional[float] = None
+    """bottom depth of the layer (mm)"""
+    nitrate: float = 1.5
+    """nitrate level of the layer ()"""  # TODO: units?
```

Comment:
The units for nitrates are *kilograms per hectare*.

```suggestion
    """nitrate level of the layer (kg/ha)"""
```

but, in addition to the volume (OK, technically this is a density at the large scale of a field) , I suspect we will also need concentration (i.e., small-scale density; see SWAT 3:1.1.1), but we can cross that bridge when we get to it.

Reply:
Ok. I would like to walk through some of the SWAT sections and figure out how you tell whether an equation is using volume, density, or concentration

Reply:
Yeah, no worries. Note that in this specific case the answer is found at the end of the first paragraph on SWAT page 333 (5:2.3.8) 

> ... and $NO3_{ly}$ is the nitrate content of soil layer $ly$ (kg $\text{NO}_{3}\text{-N}/\text{ha}$).

And the units translate to "kilograms of 'nitrate-nitrogen' (i.e., nitrogen from nitrate molecules) per hectare" and including the name of the compound in the units is redundant, so it can be reduced to (kg/ha). 

"Content" and "amount" (words that SWAT uses interchangeably) implies a quantity. In reality, though, it isn't a quantity, it as an area density (mass / area). SWAT assumes that the base unit for everything is a hectare so it uses that fact to simplify how it discusses amounts. In short: units are complicated.

That's probably more info than you needed but, there it is.

Reply:
No, that is good to know